Assets/MRTK/Tools/AssetSwapUtility/AssetSwapUtility.cs

```diff
@@ -0,0 +1,388 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using UnityEngine;
+
+#if UNITY_EDITOR
+using UnityEditor;
+#endif // UNITY_EDITOR
+
+namespace Microsoft.MixedReality.Toolkit.Utilities.Editor
+{
+    /// <summary>
+    /// A scriptable object which contains data and UI functionality to swap object (asset) references within a scene.
+    /// </summary>
+    [CreateAssetMenu(fileName = "AssetSwapCollection", menuName = "Mixed Reality Toolkit/AssetSwapCollection")]
+    public class AssetSwapUtility : ScriptableObject
+    {
+        [System.Serializable]
+        public class Theme
+        {
+            /// <summary>
+            /// The name of the theme (for the user only).
+            /// </summary>
+            public string Name;
+
+            /// <summary>
+            /// The assets that represent the theme.
+            /// </summary>
+            public List<Object> Assets;
+        }
+
+        /// <summary>
+        /// The list of themes within the collection. Must contain at least 2 items.
+        /// </summary>
+        public List<Theme> Themes = new List<Theme>();
+
+        /// <summary>
+        /// The number of assets in each theme. Must contain at least 1 item.
+        /// </summary>
+        public int AssetCount = 1;
+    }
+
+#if UNITY_EDITOR
```

Comment:
Nit, but I don't think you need these `#if`s, since the Tools asmdef is marked as editor-only:

https://github.com/microsoft/MixedRealityToolkit-Unity/blob/59a55b92903b1367190579ca2b79521ff446e228/Assets/MRTK/Tools/MRTK.Tools.asmdef#L14-L16

Reply:
Thanks for the suggestion! I had that thought and would like to opt to keep them to keep this file "portable." In case other users want to just drag and drop this tool into another project.